content/blog/2022-05-04-e2e-cv-pt2.md

```diff
@@ -0,0 +1,197 @@
+title: 'End-to-End Computer Vision API, Part 2: Local Experiments'
+date: 2022-05-04
+description: >
+  In part 1 of this series of posts, we introduced a solution to a common
+  problem faced by companies in the manufacturing industry: detecting defects
+  from images of products moving along a production line. The solution we
+  proposed was a Deep Learning-based image segmentation model wrapped in a web
+  API. We talked about effective management and versioning of large datasets and
+  the creation of reproducible ML pipelines.
+
+descriptionLong: |
+  _This is the second part of a three-part series of posts._
+
+  In part 1 of this series of posts, we introduced a solution to a common problem faced by companies in the manufacturing industry: detecting defects from images of products moving along a production line. The solution we proposed was a Deep Learning-based image segmentation model wrapped in a web API. We talked about effective management and versioning of large datasets and the creation of reproducible ML pipelines.
+
+picture: 2022-05-04/e2e-cv-pt2-cover.png
+author: alex_kim
+commentsUrl: https://discuss.dvc.org/t/end-to-end-computer-vision/1178
+tags:
+  - Computer Vision
+  - DVC
+  - CML
+  - Studio
+  - CI/CD
+  - Experiment Tracking
+---
+
+### Introduction
+
+Earlier, we built a pipeline that produces a trained Computer Vision model. Now
+we need a way to efficiently tune its configuration and the hyperparameters of
+the model. We want the ability to:
+
+- Run many experiments and easily compare their results to pick the
+  best-performing ones.
+- Track the global history of the model's performance, and map each improvement
+  to a particular change in code, configuration, or data.
+- Zoom into the details of each training run to help us diagnose issues.
+
+### Experiment Management
+
+Our DVC pipeline relies on the parameters defined in the
+[`params.yaml`](https://github.com/iterative/magnetic-tiles-defect/blob/main/params.yaml)
+file. By loading its contents in each stage, we can avoid hard-coded parameters.
+It also allows rerunning the whole or parts of our pipeline under a different
+set of parameters. The DVC pipeline YAML file
+[`dvc.yaml`](https://github.com/iterative/magnetic-tiles-defect/blob/main/dvc.yaml)
+supports a
+[templating format](https://dvc.org/doc/user-guide/project-structure/pipelines-files#templating)
+to insert values from different sources in the YAML structure itself.
+
+DVC tracks which stages of the pipeline experienced changes and only reruns
+those. By changes, we mean _everything_ that might affect the predictive
+performance of your model like changes to the dataset, source code and/or
+parameters. This not only ensures complete reproducibility but often
+significantly reduces the time needed to rerun the whole pipeline while ensuring
+consistent results on every rerun. For example, if we only change the code in
+the evaluation stage (e.g., change what type of metrics we want to compute),
```

Comment:
It might be nice to use a real example here, like why you added `dice_mean` and how it differs from other `acc_mean` or other metrics you are capturing.

Reply:
Clarified and linked to a blog post on image segmentation metrics.